aten/src/ATen/native/quantized/cpu/Pooling.cpp

```diff
@@ -232,6 +307,125 @@ Tensor q_maxpool_2d(
     return qy;
   }
 }
+
+template <typename Q>
+Tensor q_maxpool_3d(
+    Tensor qx, // Input Tensor (Quantized)
+    int64_t kT,
+    int64_t kH,
+    int64_t kW, // kernel size
+    int64_t sT,
+    int64_t sH,
+    int64_t sW, // strides
+    int64_t pT,
+    int64_t pH,
+    int64_t pW, // padding
+    int64_t dT,
+    int64_t dH,
+    int64_t dW,
+    bool ceil_mode) { // dilation
+  // Check input dimensions.
+  TORCH_CHECK(kT > 0 && kH > 0 && kW > 0, "kernel_size should be greater than zero.");
+  TORCH_CHECK(sT > 0 && sH > 0 && sW > 0, "strides should be greater than zero.");
+  TORCH_CHECK(
+      dT && dH > 0 && dW > 0,
+      "dilation should be greater than zero. "
+      "Got (",
+      dT,
+      ", ",
+      dH,
+      ", ",
+      dW,
+      ")");
+  int ndim = qx.dim();
+  TORCH_CHECK(ndim == 5, "Expecting the input tensor of rank 5.");
```

Comment:
non batch mode is not allowed ?

Reply:
Thanks for the suggestions. I add a TODO to enable the `non batch mode` for `quantized maxpool3d` when meet use cases.